src/fusor/translator.py

```diff
@@ -0,0 +1,580 @@
+        else:
+            params["structure"] = [tr_5prime[0], tr_3prime[0]]
+
+        if ce:
+            params["causativeEvent"] = ce
+        if rf:
+            params["readingFramePreserved"] = rf
+        if assay:
+            params["assay"] = assay
+        return AssayedFusion(**params), None
+
+    def _get_causative_event(
+        self, chrom1: str, chrom2: str, descr: str | None = None
+    ) -> CausativeEvent | None:
+        """Infer Causative Event. Currently restricted to rearrangements
+
+        :param chrom1: The chromosome for the 5' partner
+        :param chrom2: The chromosome for the 3' partner
+        :param descr: An annotation describing the fusion event. This input is supplied to the eventDescription CausativeEvent attribute.
+        :return: A CausativeEvent object if construction is successful
+        """
+        if descr and "rearrangement" in descr:
+            return CausativeEvent(
+                eventType=EventType("rearrangement"), eventDescription=descr
+            )
+        if chrom1 != chrom2:
+            return CausativeEvent(eventType=EventType("rearrangement"))
+        return None
+
+    def _get_gene_element_unnormalized(self, symbol: str) -> GeneElement:
+        """Return GeneElement when gene symbol cannot be normalized
+
+        :param symbol: A gene symbol for a fusion partner
+        :return: A GeneElement object
+        """
+        return GeneElement(
+            type="GeneElement",
+            gene={
+                "id": f"gene:{symbol}",
+                "label": symbol,
+                "type": "Gene",
+            },
+        )
+
+    def _get_gene_element(self, genes: str, caller: Caller) -> GeneElement:
+        """Return a GeneElement given an individual/list of gene symbols and a
+        fusion detection algorithm
+
+        :param genes: A gene symbol or list of gene symbols, separated by columns
+        :param caller: The examined fusion detection algorithm
+        :return A GeneElement object
+        """
+        if "," not in genes or caller != caller.ARRIBA:
+            ge = self.fusor.gene_element(gene=genes)
+            return ge if ge[0] else self._get_gene_element_unnormalized(genes)
+
+        genes = genes.split(",")
+        dists = []
+        for gene in genes:
+            start, end = gene.rfind("("), gene.rfind(")")
+            dists.append(int(gene[start + 1 : end]))
+        gene = (
+            genes[0].split("(")[0] if dists[0] <= dists[1] else genes[1].split("(")[0]
+        )
+        ge = self.fusor.gene_element(gene=gene)
+        return ge if ge[0] else self._get_gene_element_unnormalized(gene)
+
+    def _get_genomic_ac(self, chrom: str, build: Assembly) -> str:
+        """Return a RefSeq genomic accession given a chromosome and a reference build
+
+        :param chrom: A chromosome number
+        :param build: The assembly, either GRCh37 or GRCh38
+        :return The corresponding refseq genomic accession
+        """
```

Comment:
you raise a ValueError so you need to document it in the docstring ie

```
:raise ValueError: if unable to retrieve genomic accession
```

Reply:
the reason to format like this is it means we can automatically generate documentation that lists the kinds of exceptions a user needs to guard against, eg

https://coolseqtool.readthedocs.io/0.8.0/reference/api/sources/cool_seq_tool.sources.uta_database.html#cool_seq_tool.sources.uta_database.UtaDatabase.get_secret